content/filterx/_index.md

```diff
@@ -0,0 +1,377 @@
+
+<!-- FIXME more examples for lists/dicts if needed -->
+
+<!--
+### Type casting
+
+ FIXME type casting -->
+
+## Operators
+
+Filterx has the following operators.
+
+- Comparison operators: `==, <, <=, >=, >, !=, eq, lt, le, gt, ge, ne`. For details, see {{% xref "/filterx/filterx-comparing/_index.md" %}}.
+- Boolean operators `not`, `or`, `and`. For details, see {{% xref "/filterx/filterx-boolean/_index.md" %}}.
+- [Dot operator (`.`)](#json) to access fields of an object, like JSON.
+- [Indexing operator `[]`](#json) to access fields of an object, like JSON.
+- [Plus (`+`) operator](#concatenate-strings) to concatenate strings.
+- [Ternary conditional operator]({{< relref "/filterx/operator-reference.md#ternary-conditional-operator" >}}): `?:`
+- [Null coalescing operator]({{< relref "/filterx/operator-reference.md#null-coalescing-operator" >}}): `??`
+- [Regular expression (regexp) match]({{< relref "/filterx/operator-reference.md#regexp" >}}): `=~` and `!~`.
+
+<!--  +=  -->
+
+<!-- FIXME 
+`$MSG = "$(format-json --subkeys values.)";` > needs the double-quote for template eval > should we document that Bazsi?
+-->
+
+For details, see {{% xref "/filterx/operator-reference.md" %}}.
+
+## Functions
+
+Filterx has the following built-in functions.
+
+<!-- FIXME definitions/descriptions -->
+- [`cache_json_file`]({{< relref "/filterx/function-reference.md#cache_json_file" >}})
+- [`datetime`]({{< relref "/filterx/function-reference.md#datetime" >}})
+- [`flatten`]({{< relref "/filterx/function-reference.md#flatten" >}})
+- [`isodate`]({{< relref "/filterx/function-reference.md#isodate" >}})
+- [`isset`]({{< relref "/filterx/function-reference.md#isset" >}})
+- [`istype`]({{< relref "/filterx/function-reference.md#istype" >}}): Check the type of an object.
+- [`len`]({{< relref "/filterx/function-reference.md#len" >}}): Returns the length of an object.
+- [`lower`]({{< relref "/filterx/function-reference.md#lower" >}}): Converts a string into lowercase characters.
+- [`json, json_object`]({{< relref "/filterx/function-reference.md#json" >}}): Convert a value into a JSON object.
+- [`json_array`]({{< relref "/filterx/function-reference.md#json-array" >}}): Convert a value into a JSON array.
+- [`string`]({{< relref "/filterx/function-reference.md#string" >}}): Convert a value into a string.
+- [`strptime`]({{< relref "/filterx/function-reference.md#strptime" >}}): Convert a value into datetime.
+- [`unset`]({{< relref "/filterx/function-reference.md#unset" >}}): Delete a name-value pair, or a field from an object.
+- [`unset_empties`]({{< relref "/filterx/function-reference.md#unset-empties" >}}): Delete empty fields from an object.
+- [`upper`]({{< relref "/filterx/function-reference.md#upper" >}}): Converts a string into uppercase characters.
+- [`vars`]({{< relref "/filterx/function-reference.md#vars" >}}): The variables defined in the filterx block.
+- "bytes"
+- "protobuf"
+- "bool"
+- "int"
+- "double"
```

Comment:
we may handle these separately (not in this list), they are already mentioned for explicit typing